samples: drivers: watchdog: Update sample to use alias

benjaminbjornsson opened the pull request:
Simplify sample by using alias for node access. Also removes
usage of DT_LABEL.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

martinjaeger commented on samples/drivers/watchdog/src/main.c:
```diff
@@ -14,39 +14,12 @@
-#define WDT_NODE DT_ALIAS(watchdog0)
-#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32_window_watchdog)
-#define WDT_NODE DT_INST(0, st_stm32_window_watchdog)
-#define WDT_MAX_WINDOW  100U
```
This special case still needs to be considered.

benjaminbjornsson replied:
Oh, I didn’t see that one.. thanks!

gmarull replied:
this is a sympthom that we likely need a watchdog specifier, like we have e.g., for adc or pwm.

martinjaeger reviewed (commented):
With this change many boards won't work for this sample anymore as they don't have the alias defined. I suggest to add the alias to all boards which have `status = "okay"` for their watchdog in their board `.dts` file and not only in the overlays.

benjaminbjornsson commented:
> With this change many boards won't work for this sample anymore as they don't have the alias defined. I suggest to add the alias to all boards which have `status = "okay"` for their watchdog in their board `.dts` file and not only in the overlays.

Great suggestion, will update!

benjaminbjornsson committed:
boards: arm: cc32xxsf_launchxl: Enable watchdog in devicetree.

Enable watchdog in device tree in order to remove overlays
from samples/drivers/watchdog.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
samples: drivers: watchdog: Remove uneccesary overlays

These watchdogs are already enabled in device tree.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
boards: Add alias to boards with watchdog enabled

Add alias to boards with watchdog enabled to facilitate the
move of samples/drivers/watchdog to use DT_ALIAS.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
samples: drivers: watchdog: Add alias to overlays

Add alias to overlays for boards that switch the watchdog to be
used while running this sample.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
samples: drivers: watchdog: Update sample to use alias

Simplify sample by using DT_ALIAS to make the sample more generic.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

benjaminbjornsson committed:
boards: arm: efm32pg_stk340xa: Move watchdog from chosen to alias

Move watchdog0 and watchdog1 nodes from choosen to aliases for
consistency with other boards.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

martinjaeger reviewed (approved):
Thanks for the update, LGTM.
